src/python/pants/goal/run_tracker.py

```diff
@@ -390,3 +395,32 @@ def shutdown_worker_pool(self):
     N.B. This exists only for internal use and to afford for fork()-safe operation in pantsd.
     """
     SubprocPool.shutdown(self._aborted)
+
+  def report_target_info(self, scope, target, key, val):
+    """Add target information to run_info under target_data.
+
+    Will Recursively construct a nested dict with the keys provided
+
+    :param string scope: The running scope
```

Comment:
Nit: "running scope" isn't really a term of art. How about "The scope we're reporting information for."?